### ISSUE TITLE
[Improvement] -Add key info to Concierge to improve the New Expensify customer support experience

### ISSUE THREAD

**Issue by VictoriaExpensify** [Expensify/App#48529]
**Proposal**
Include thread and task details when a customer reaches out to Concierge from a threaded message 

**Problem**
As discussed [here](https://expensify.slack.com/archives/C07HPDRELLD/p1725029615796859) and [here](https://expensify.slack.com/archives/C041MSTR8J1/p1723823732143719), customers can now reach out to Concierge in task threads; however, Concierge does not have visibility of the tasks the user is writing in from. This means that Concierge lacks the context needed to support the customer; this is resulting in an awkard customer experience because Concierge needs to get clarification on details that the customer would expect us to be able to see.

**Solution**
When a customer reaches out to Concierge within a thread, make sure the responding agent has visibility of the information they need to assist that customer. This includes:
- Showing details of the parent thread and header the customer is reaching out from
- Show reactions to messages (e.g. if a customer adds a 👍 to a message, ensure this is shown in Concierge)

## [Design doc](https://docs.google.com/document/d/1pr7rcJKbgwYvQIuUtpobZqxkCeNMi6Gnt_bSnBhEzUs/edit?usp=gmail#heading=h.mnjguil21u0a) (Detailed portion currently in review)

**Comment by VictoriaExpensify** [comment]
Proposal taken to WN - plenty of support for this and no pushback - https://expensify.slack.com/archives/CC7NECV4L/p1725492990710679

**Comment by VictoriaExpensify** [comment]
Ok this is actually going to go through a pre-design. Will start drafting this today

**Comment by VictoriaExpensify** [comment]
**Update Sept 11:**
Pre-design is being drafted with this and will be posted in Concierge Development later in the week

**Comment by VictoriaExpensify** [comment]
Pre-design posted - https://expensify.slack.com/archives/C041MSTR8J1/p1726176871367989

**Comment by VictoriaExpensify** [comment]
Pre-design has been wrapped up and I will start work finding an engineer to help implement

**Comment by muttmuure** [comment]
Is this ready for the label?

**Comment by VictoriaExpensify** [comment]
As discussed in the [Pre-Design](https://expensify.slack.com/archives/C041MSTR8J1/p1726176871367989), this is what needs to be included in Concierge:

**1. [Onboarding tasks checklist when the customer replies to Concierge](https://expensify.slack.com/archives/C041MSTR8J1/p1726177020394499)**
This refers to the onboarding checklist that the customer sees in the New Expensify chat with Concierge:

![image](https://github.com/user-attachments/assets/dad88ee1-c7bf-4595-97a3-ab25203aefba)
This is not currently shown in Concierge:
![image](https://github.com/user-attachments/assets/ea2f605b-e12d-4926-8ff6-99660d138bb2)

The list should show the tasks a customer has completed, either through a check or a strikethrough - whichever solution is most simple to implement

**2. [The parent thread and header when the customer summons Concierge from within an onboarding task thread](https://expensify.slack.com/archives/C041MSTR8J1/p1726177453130969)**

- When a customer starts a new thread, in Concierge we need to show the details of the message the customer is responding to
- For example, this is the parent thread https://www.expensify.com/concierge/#/chat/20623863 with Concierge
I've responded to the Meet your setup specialist task, starting a new thread and a [new concierge chat](https://www.expensify.com/concierge/#/chat/20623899)
- The new chat that was started in NewDot looks like this:
![image](https://github.com/user-attachments/assets/42aee600-2ff3-46cd-8b6c-b51e293a1980)

-   should look something like this:
![image](https://github.com/user-attachments/assets/8b02fb01-2cf3-4028-88e1-a39610e72381)

**[3. Reference the main thread chat within the sub thread chat when a customer starts a new chat by replying in thread](https://expensify.slack.com/archives/C041MSTR8J1/p1726460716756249?thread_ts=1726177583.448579&cid=C041MSTR8J1)**

- When a customer starts a new thread, which will automatically create a new Concierge chat, the parent thread should show a link to the chat created from the  new thread
- Likewise, the chat created from the new thread should show a link to the parent thread
- This would look similar to what we use today when new people are cc'd into a response (see screenshot)

![image](https://github.com/user-attachments/assets/b1b8754c-58bf-4579-84d3-5332ea675c54)


For example, https://www.expensify.com/concierge/#/chat/20623863 is the main Concierge chat thread for vik.oleary+threadstest@gmail.com. This user replied in thread and a new chat was created https://www.expensify.com/concierge/#/chat/20623951

The new/sub chat thread (https://www.expensify.com/concierge/#/chat/20623951) should reference the original/main chat thread (https://www.expensify.com/concierge/#/chat/20623863) and vice-versa

[4. Show reactions to Concierge messages](https://expensify.slack.com/archives/C041MSTR8J1/p1726177712737239)
At the moment, if a customer reacts to a Concierge message with an emoji, Concierge does not see this or get notified.

- Emoji reactions should be sent to Concierge as a new message, along with the context of the message the customer is reacting to (because to may not be the most recent)
- If the chat is closed, the emoji reaction should trigger the chat to be reopened

**Comment by VictoriaExpensify** [comment]
Chatted to @marcaaron - he's about to go OOO so isn't going to be able to get to this for several weeks. Adding the auto-assigner again

**Comment by deetergp** [comment]
@VictoriaExpensify I've already promised @alexpensify that I would work with him on the technical portion of Vacation Delegates 😬

**Comment by VictoriaExpensify** [comment]
Well darn!

Ok I'm going to throw a Design Doc together for this anyway. Once that's done and reviewed, maybe Marc will nearly be back from leave or you will nearly be done with the Vacation Delegates project.

If not, I'll spin the wheel again - third time lucky, right? 😅

**Comment by marcaaron** [comment]
Ooh might have better luck asking around vs. auto assigning for a Design Doc! I'm just thinking the odds could be high that your auto-assignee is already be working on something big, but it's worth a shot.

**Comment by deetergp** [comment]
Agreed… Asking around is going to be your best bet.

**Comment by VictoriaExpensify** [comment]
Thanks team! I'm chatting with Chirag about this at the moment 🤞

**Comment by VictoriaExpensify** [comment]
Not OD, Chirag and I are working on this - https://docs.google.com/document/d/19xghDYgBkMgQUBKkNqO2WeayMgKrsut2yinVxi3y3rg/edit

**Comment by chiragsalian** [comment]
Working on this. We just posted a predesign discussion [here](https://expensify.slack.com/archives/C041MSTR8J1/p1727895327674569).

**Comment by chiragsalian** [comment]
Design doc is ready for high level review here - https://docs.google.com/document/d/1pr7rcJKbgwYvQIuUtpobZqxkCeNMi6Gnt_bSnBhEzUs/edit?usp=gmail#heading=h.mnjguil21u0a

**Comment by tgolen** [comment]
I have read and reviewed this Design Doc!

**Comment by trentpetty** [comment]
I have read and reviewed this Design Doc!

**Comment by aldo-expensify** [comment]
I have read and reviewed this Design Doc!

**Comment by stephanieelliott** [comment]
I have read and reviewed this Design Doc!

**Comment by jamesdeanexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by VictoriaExpensify** [comment]
**Timeline and urgency**
- The limitation in the information Concierge is already having an impact on the support experience our customers are receiving ([example](https://expensify.slack.com/archives/C07HPDRELLD/p1728265141124089))
- This is [now a critical issue](https://expensify.slack.com/archives/C07HPDRELLD/p1728265141124089) and it should be treated as a top priority

**Comment by tgolen** [comment]
I got assigned to a detailed review, but I see the high-level part of the doc hasn't completely been reviewed yet and there doesn't appear that the detailed part of the doc has been written.

**Comment by VictoriaExpensify** [comment]
Oh this is my fault! If you were auto-assigned for a detailed review, please ignore this 😬

**Comment by flaviadefaria** [comment]
I have read and reviewed this Design Doc!

**Comment by garrettmknight** [comment]
I have read and reviewed this Design Doc!

**Comment by chiragsalian** [comment]
Oh looks like designDocReview was added a second time incorrectly here. @VictoriaExpensify, I'm going to delete the message,
![image](https://github.com/user-attachments/assets/aa48793b-db14-4948-8aec-e931c90a63ec)

and close out these issues. 
![image](https://github.com/user-attachments/assets/d9f11cb0-18c4-4e8b-b178-85b2be47995d)

As they are incorrect dupes, because we have the above issues and message for high level review. 
![image](https://github.com/user-attachments/assets/4b0b5f70-c89b-43c9-83e1-d5d38a7fc124)

**Comment by chiragsalian** [comment]
Done clearing up the recent assignment so that people don't waste time since we already have assignees for the high level review.

**Comment by RachCHopkins** [comment]
I have read and reviewed this Design Doc!

**Comment by tylerkaraszewski** [comment]
I have read and reviewed this Design Doc!

**Comment by chiragsalian** [comment]
Detailed portion is ready for review. Woop woop.

**Comment by flaviadefaria** [comment]
I have read and reviewed this Design Doc!

**Comment by tgolen** [comment]
I have read and reviewed this Design Doc!

**Comment by jamesdeanexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by nkuoch** [comment]
I have read and reviewed this Design Doc!

**Comment by garrettmknight** [comment]
I have read and reviewed this Design Doc!

**Comment by VictoriaExpensify** [comment]
@aldo-expensify and @zsgreenwald we're just waiting on your reviews when you have a moment. Thanks!

**Comment by marcaaron** [comment]
I have read and reviewed this Design Doc!

**Comment by VictoriaExpensify** [comment]
Reviews are complete, working on it

**Comment by VictoriaExpensify** [comment]
In progress, not OD

**Comment by VictoriaExpensify** [comment]
As above ^^

**Comment by JmillsExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by VictoriaExpensify** [comment]
Not Overdue, @chiragsalian is making solid progress:

**Complete:**
- [ ] [PHP] Create method syncReportResponsesToChatBot in ChatBot Utils #440661
- [ ] [Auth] Add a new auth command - GetReportParentID #440664
- [ ] [PHP] Update createChatbotInputFromReportComment to post parent/child notes #440676

From a responder perspective, we are starting to see these improvements which is helping to improve our interactions:
![image](https://github.com/user-attachments/assets/11a4b175-1e6c-4e5d-a06b-0d4d69d784c4)

**In progress:**
- [ ] [Auth] Update OpenReport auth to share Concierge threads with Concierge #440663
- [ ] [Auth] Updating addEmojiReaction to identify user reactions to Concierge responses #440677
- [ ] [PHP] Update addEmojiReaction to post a note for user reaction #440678

**On hold:**
The following are on hold as the Try New Expensify button has recently been added to product and we need to confirm the new behaviour/routing with that button:
- [ ] [Auth][HOLD] New NVP recentlyUsedApp #440679
- [ ] [HOLD][PHP] Set supporting source when support in oldDot is clicked #440681
- [ ] [HOLD] [PHP] Set chatInput supporting source for incoming user messages #440715
- [ ] [HOLD] [PHP] Present the user with a whisper message indicating the focus of support #440722
- [ ] [HOLD] [Web] Display recentlyUsedApp in user info and allow it to be modified by agents to better support their customers #440723
- [ ] [HOLD] [PHP] New API endpoint ChatBot_User_SetRecentlyUsedApp #440724

I will post a detailed update to #retain tomorrow

**Comment by VictoriaExpensify** [comment]
@chiragsalian I will start playing around with the "Try New Expensify" button tomorrow and get some logs and NVP behaviour for you

**Comment by chiragsalian** [comment]
Not overdue, finished all the pieces. The last one got merged today and should go live end of this week or early next week and then we can close out this issue.

**Comment by VictoriaExpensify** [comment]
Ok so let's run through the changes that have been made here:

**1. Showing the Onboarding tasks checklist when the customer sends a reply to the Concierge DM ([discussion](https://expensify.slack.com/archives/C041MSTR8J1/p172617702039449))** 👀 

I can't test this because I have a bug which leads to onboarding tasks not being created (posted [here](https://expensify.slack.com/archives/C05LX9D6E07/p1733121675571129)), but I have seen this live in Concierge. I will check again tomorrow.

This change was not retroactive so I can't use an existing account to check this

**2. Showing details of the threaded message the customer is replying to in the Concierge DM ([discussion](https://expensify.slack.com/archives/C041MSTR8J1/p1726177453130969)) - parent chat view** ✅ 

When I started a thread from a task, a note was added to my main Concierge chat with a link to the new threaded chat:
![image](https://github.com/user-attachments/assets/ea2112cb-eb1f-4f3f-8967-7d45ea5372cd)

The threaded chat also contains a link to the main Concierge chat:
![image](https://github.com/user-attachments/assets/5eedb320-9b3b-4e75-9ab2-74efc82285a0)

Confirmed these links routed to the correct chats 👍 

**3. Reference the main Concierge Chat when the customer starts a new thread in New Expensify ([discussion](https://expensify.slack.com/archives/C041MSTR8J1/p1726460716756249?thread_ts=1726177583.448579&cid=C041MSTR8J1)) - child chat view**  ✅ 

When a customer responds to Concierge in thread, a new Concierge chat is created and this chat is now referenced in the main Concierge chat:
![image](https://github.com/user-attachments/assets/baf8064e-7f1e-423e-866f-c6d519e0aac6)

![image](https://github.com/user-attachments/assets/b5d2e6b7-0619-47d5-9df2-94922e5fd9b4)

The newly created chat also references the main chat:
![image](https://github.com/user-attachments/assets/be7b50f0-09d2-481d-9241-8595ca0659bf)

Confirmed these links routed to the correct chats 👍 

**4. Show reactions to Concierge messages ([discussion](https://expensify.slack.com/archives/C041MSTR8J1/p1726177712737239)1, [discussion2](https://expensify.slack.com/archives/C041MSTR8J1/p1727895432308929))**

I think this is the final piece you're working on, is that right @chiragsalian ?

5. Add a new identifier to differentiate if a user is writing about a query with oldDot or newDot ([discussion](https://expensify.slack.com/archives/C041MSTR8J1/p1727895473761559))

_Solution:
Create another NVP, we’ll call this “recentlyUsedApp”.
This NVP:
Is set to “oldDot”, 
- When the user logs into Expensify Classic
- When the user clicks on Support->Concierge in Expensify Classic_

Testing with victoria@expensifail.com which is predominantly used with Classic:
- Signed into Classic
- Reached out to Concierge via Support > Concierge (which redirects me to New Expensify):
![image](https://github.com/user-attachments/assets/abac9a43-4fa1-49a1-9b64-5812601b9ff2)

This should show that the user is Classic so it isn't working as expected. @chiragsalian before I go ahead with more testing for the NVP, this change should be in place now, correct?

**Comment by VictoriaExpensify** [comment]
I need to do some more testing on the NVPs, will provide further updates soon

**Comment by twisterdotcom** [comment]
I am removing hot pick from any issues with an internal engineer assigned so other engineers can know it's available.

**Comment by VictoriaExpensify** [comment]
Thanks Ted

**Comment by anmurali** [comment]
@VictoriaExpensify - is there an update on the testing?

**Comment by VictoriaExpensify** [comment]
Yeah, there are issues with the NVP for existing Classic users - it only displays `NVP: NONE` and this doesn't update when switching between New and Classic. Testing details here https://docs.google.com/document/d/1rFUHVSAHWCiMnQJ5x2IYjZiNDc5CSd54bIn-SSCI8EI/edit?tab=t.0

More info to come

**Comment by VictoriaExpensify** [comment]
The NVP seems to update well when the "Try New Expensify" button is used, so that's a win. But if someone signs into New Expensify directly, it is inaccurate.

**Comment by VictoriaExpensify** [comment]
We're working on a plan for updated NVPs. More info coming!

**Comment by chiragsalian** [comment]
I need to post a plan for the updated NVps. Haven't gotten around to it just yet but ill post it hopefully soon.

**Comment by VictoriaExpensify** [comment]
I'm just back from ooo but Chirag and I are catching up tomorrow to make a plan here

**Comment by VictoriaExpensify** [comment]
Just adding a couple of items to this that have come up in Slack discussions, so we don't forget:

- Remove the New/Classic message indicator (all users get redirected to New Expensify to message Concierge now, so this is redundant). We should keep the email flag
- Remove the “user switched to classic” messaging (this conflicts with the NVP in the User Info tab)
- User Info NVP = source of truth

**Comment by VictoriaExpensify** [comment]
Chirag has tested the improvements and is working on implementing them - https://github.com/Expensify/Expensify/issues/459152

**Comment by VictoriaExpensify** [comment]
Updates to the NVP are now in staging - woohoo! We are this 🤏 close to closing this out. Working on the final update

**Comment by nikihatesgh** [comment]
@VictoriaExpensify @chiragsalian as per the accounting review section of the design doc this project was not eligible for cap sw time tracking. I cannot find the GH request to get it added to the cap sw sheet but can each of you please acknowledge that for future projects you will only action the cap sw request after the accounting review confirms it is eligible? 

![Image](https://github.com/user-attachments/assets/c301a02b-92e6-4f0f-83a5-f17291c8be87)

**Comment by VictoriaExpensify** [comment]
Hi @nikihatesgh, sorry, I misunderstood the process. I got assigned [this issue](https://github.com/Expensify/Expensify/issues/449302) from Accounting and assumed that meant they were signing it off as a CAP issue. But now I see Trent's comments. I assume that GH was triggered because the project got added to the CAP spreadsheet at some point.

**Comment by VictoriaExpensify** [comment]
The flow looks really good and, from testing, we are accurately capturing:
- Classic users signing into Classic and messaging Concierge
- New Expensify users signing into New Expensify  and messaging Concierge
- Classic users that use the Try New Expensify button to switch
- New users that use the Switch to Classic button to switch back

The scenarios that are not being captured are:
- Classic users who sign into New Expensify directly - I think this is only a small issue for now but could this grow as the New Expensify URL and App are in circulation more?
- Customer that write in from email - at the moment, they do not show New or Old, just Email. It would be good to show both Email and the platform they have been using most recently

**Comment by chiragsalian** [comment]
Not overdue, we're talking about it in chat.
> Classic users who sign into New Expensify directly - I think this is only a small issue for now but could this grow as the New Expensify URL and App are in circulation more?

imo, if we've solved for the majority of the cases we should close out this project and create separate GH issues for fine tuning the remaining items.

> Customer that write in from email - at the moment, they do not show New or Old, just Email. It would be good to show both Email and the platform they have been using most recently

Yup both are shown. One is shown below the message and the other is in user info.

**Comment by VictoriaExpensify** [comment]
And we're done! Our final wrap up post is here - https://expensify.slack.com/archives/C07NZ8B1VTQ/p1737685167553009
